From 83cb24c21cd1fcd2009572219a58d6b2837b243c Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Wed, 29 Oct 2014 16:34:16 -0400
Subject: [fs] ext4: remove a duplicate call in ext4_init_new_dir()

Message-id: <1414600485-1929-27-git-send-email-lczerner@redhat.com>
Patchwork-id: 98544
O-Subject: [RHEL 7.1 PATCH v2 26/55] ext4: remove a duplicate call in ext4_init_new_dir()
Bugzilla: 1156096
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

From: Wang Shilong <wshilong@ddn.com>

BZ 1156096
https://bugzilla.redhat.com/show_bug.cgi?id=1156096

BREW 8169722
https://brewweb.devel.redhat.com/taskinfo?taskID=8169722

Upstream commit 52c826db6d4b638677683c79e6c465b99074be74

 ext4_journal_get_write_access() has just been called in ext4_append()
 calling it again here is duplicated.

 Signed-off-by: Wang Shilong <wshilong@ddn.com>
 Signed-off-by: Theodore Ts'o <tytso@mit.edu>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/namei.c b/fs/ext4/namei.c
index 1d2c333..d12a255 100644
--- a/fs/ext4/namei.c
+++ b/fs/ext4/namei.c
@@ -2365,10 +2365,6 @@ static int ext4_init_new_dir(handle_t *handle, struct inode *dir,
  dir_block = ext4_append(handle, inode, &block);
  if (IS_ERR(dir_block))
   return PTR_ERR(dir_block);
- BUFFER_TRACE(dir_block, "get_write_access");
- err = ext4_journal_get_write_access(handle, dir_block);
- if (err)
-  goto out;
  de = (struct ext4_dir_entry_2 *)dir_block->b_data;
  ext4_init_dot_dotdot(inode, de, blocksize, csum_size, dir->i_ino, 0);
  set_nlink(inode, 2);
-- 
1.7.1